Lib/ufo2ft/__init__.py

```diff
@@ -600,48 +641,155 @@ def compileVariableCFF2(designSpaceDoc, **kwargs):
+
+    *variableFontNames* is an optional list of filenames of variable fonts
+      to build. If not provided, all variable fonts listed in the given
+      designspace will by built.
+
+    The rest of the arguments works the same as in the other compile functions.
+
+    Returns a dictionary that maps each variable font filename to a new variable
+    TTFont object.
+
+    .. versionadded:: TODO (Jany) version of addition
+    """
+    kwargs = init_kwargs(kwargs, compileVariableCFF2s_args)
+    excludeVariationTables = kwargs.pop("excludeVariationTables")
     optimizeCFF = CFFOptimization(kwargs.pop("optimizeCFF"))
+    variableFontNames = kwargs.pop("variableFontNames")
 
-    varfont = varLib.build(
-        otfDesignSpace,
+    # Pop inplace because we'll make a copy at this level so deeper functions
+    # don't need to worry
+    inplace = kwargs.pop("inplace")
+    if not inplace:
+        designSpaceDoc = designSpaceDoc.deepcopyExceptFonts()
+
+    vfNameToBaseUfo = _compileNeededSources(
+        kwargs, designSpaceDoc, variableFontNames, compileInterpolatableOTFsFromDS
+    )
+
+    if not vfNameToBaseUfo:
+        logger.warning("No variable fonts to build")
+        return {}
+
+    logger.info(f"Building variable CFF2 fonts: {', '.join(vfNameToBaseUfo)}")
+
+    vfNameToTTFont = varLib.build_many(
+        designSpaceDoc,
         exclude=excludeVariationTables,
         # NOTE optimize=False won't change anything until this PR is merged
         # https://github.com/fonttools/fonttools/pull/1979
         optimize=optimizeCFF >= CFFOptimization.SPECIALIZE,
-    )[0]
-
-    return call_postprocessor(
-        varfont,
-        baseUfo,
-        glyphSet=None,
-        **kwargs,
-        optimizeCFF=optimizeCFF >= CFFOptimization.SUBROUTINIZE,
+        skip_vf=lambda vf_name: variableFontNames and vf_name not in variableFontNames,
     )
+
+    for vfName, varfont in list(vfNameToTTFont.items()):
+        vfNameToTTFont[vfName] = call_postprocessor(
+            varfont,
+            vfNameToBaseUfo[vfName],
+            glyphSet=None,
+            **kwargs,
+            optimizeCFF=optimizeCFF >= CFFOptimization.SUBROUTINIZE,
+        )
+
+    return vfNameToTTFont
+
+
+def _compileNeededSources(
+    kwargs, designSpaceDoc, variableFontNames, compileInterpolatableFunc
+):
+    # We'll need to map <source> elements to TTFonts, to do so make sure that
+    # each <source> has a name.
+    ensure_all_sources_have_names(designSpaceDoc)
+
+    # Go through VFs to build and gather list of needed sources to compile
+    interpolableSubDocs = [
+        subDoc for _location, subDoc in splitInterpolable(designSpaceDoc)
+    ]
+    vfNameToBaseUfo = {}
+    sourcesToCompile = set()
+    for subDoc in interpolableSubDocs:
+        for vfName, vfDoc in splitVariableFonts(subDoc):
+            if variableFontNames is not None and vfName not in variableFontNames:
```

Comment:
make sure variableFontNames is initialized to an empty set so don't need to check every time if it's not None

Reply:
actually discard my silly comment 😅